core/include/detray/tools/bin_association.hpp

```diff
@@ -194,7 +191,8 @@ static inline void bin_association(const context_t & /*context*/,
                                 -std::numeric_limits<scalar>::max();
 
                             for (const auto &v : vertices) {
-                                const point3 vg = transform.point_to_global(v);
+                                const point3_t vg =
+                                    transform.point_to_global(v);
```

Comment:
Are vertices created from the `generators`? In that case, `point_to_global` of transform3 would not produce the right global position. (It's only for the cartesian coordinate). We might need to use the `local_to_global` function that I added in algebra-plugins recently. Of course it's not the scope of this PR so we can fix it later.

Reply:
I am not sure, tbh :D. I will properly understand that code, once I move to filling the grids, I guess